tests/events/client/instrumentation/test_task_run_state_change_events.py

```diff
@@ -36,44 +39,174 @@ def happy_path():
     ]
     assert len(task_run_states) == len(events) == 3
 
-    last_state = None
-    for i, task_run_state in enumerate(task_run_states):
-        event = events[i]
```

Comment:
Note: I found this form of the tests a little hard to reason about, so I have unrolled and flattened them out into mostly hard-coded values